File: MPI_Code/C/hello_host.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char *argv[])
{
   int ver,subver;
   int nRank, nProcs;
   char procName[MPI_MAX_PROCESSOR_NAME];
   int nNameLen;
   

   MPI_Init(NULL,NULL);
   MPI_Comm_rank(MPI_COMM_WORLD,&nRank);
   MPI_Comm_size(MPI_COMM_WORLD,&nProcs);

   MPI_Get_version(&ver,&subver);
   if(nRank==0)printf("MPI Version %d.%d\n",ver,subver);
   MPI_Get_processor_name(procName, &nNameLen);
    
   printf("Hello World.(Process name=%s, nRank=%d, nProcs=%d)\n",procName, nRank, nProcs);
   MPI_Finalize();
   return 0;
}

File: example_code/mpi/MPI_Code/Fortran/hello_host.f90
PROGRAM hello
USE mpi_f08
IMPLICIT NONE
INTEGER::nRank, nProcs, nNameLen
CHARACTER(LEN=MPI_MAX_PROCESSOR_NAME)::procName
INTEGER::ver, subver

CALL MPI_Init
CALL MPI_Comm_size(MPI_COMM_WORLD, nProcs)
CALL MPI_Comm_rank(MPI_COMM_WORLD, nRank)

CALL MPI_Get_version(ver,subver)
IF(nRank==0) PRINT*,'MPI Version',ver,'.',subver
CALL MPI_Get_processor_name(procName,nNameLen)
PRINT*,'Hello World.(Process name=',TRIM(procName),', nRank=',nRank,',nProcs=',nProcs
CALL MPI_Finalize
END PROGRAM hello
